master/internal/api_command.go

```diff
@@ -75,6 +70,15 @@ func (a *apiServer) getCommandLaunchParams(ctx context.Context, req *protoComman
 			status.Errorf(codes.Unauthenticated, "failed to get the user: %s", err)
 	}
 
+	cmdSpec.Metadata.WorkspaceID = model.DefaultWorkspaceID
+	if req.WorkspaceID != 0 {
+		cmdSpec.Metadata.WorkspaceID = model.AccessScopeID(req.WorkspaceID)
+		// ensure that workspace name is propagated to task spec when specified via CLI
+		if w, _ := a.GetWorkspaceByID(ctx, req.WorkspaceID, *userModel, true); w != nil {
+			cmdSpec.Base.Workspace = w.Name
```

Comment:
This is an attempt to get the workspace name propagated to the task spec, but it isn't working for notebooks.